kola/tests/misc/nvidia.go

```diff
@@ -0,0 +1,52 @@
+package misc
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/flatcar/mantle/kola/cluster"
+	"github.com/flatcar/mantle/kola/register"
+	"github.com/flatcar/mantle/util"
+)
+
+const (
+	CmdTimeout = time.Second * 300
+)
+
+func init() {
+	register.Register(&register.Test{
+		Name:        "cl.misc.nvidia",
+		Run:         verifyNvidiaInstallation,
+		ClusterSize: 1,
+		Distros:     []string{"cl"},
+		// This test is to test the NVIDIA installation, limited to AZURE for now
+		Platforms:     []string{"azure"},
+		Architectures: []string{"amd64"},
+	})
+}
+
+func verifyNvidiaInstallation(c cluster.TestCluster) {
+	m := c.Machines()[0]
+	nvidiaStatusRetry := func() error {
+		out, err := c.SSH(m, "systemctl status nvidia.service")
+		if err != nil {
+			return err
+		}
+
+		if !bytes.Contains(out, []byte("Active: inactive")) {
+			return fmt.Errorf("nvidia.service: %q: %v", out, err)
+		}
+
+		return nil
+	}
+
+	if err := util.Retry(20, 15*time.Second, nvidiaStatusRetry); err != nil {
+		c.Fatal(err)
+	}
+
+	out, err := c.SSH(m, "/opt/bin/nvidia-smi")
+	if !bytes.Contains(out, []byte("525.105.17")) {
```

Comment:
This is going to require maintenance at every release - can the version be queried from flatcar metadata in the image? Or maybe just check that nvidia-smi succeeds, and shows a GPU.